### PR TITLE
Fix share post in institution page

### PR DESCRIPTION
**Pull request by MaianaB**
Bug description: When share post in page institution, the shared post is added in timeline of institution even if institution's page not be institution author of post.

Solution: Verify if user is in page institution

### CODE REVIEW

File: app/post/postDetailsDirective.js [hunk 1]
```diff
@@ -620,13 +622,19 @@
...
-                posts.push(response.data);
+                shareCtrl.addPostTimeline(response.data);
             }, function error(response) {
                 $mdDialog.hide();
                 MessageService.showToast(response.data.msg);
             });
         };
 
+        shareCtrl.addPostTimeline = function addPostTimeline(post) {
+            if(!institution || institution.key == user.current_institution.key){
```
Comment (Luiz-FS) [inline app/post/postDetailsDirective.js]: Extrair para constante.

File: app/post/postDetailsDirective.js [hunk 2]
```diff
@@ -581,7 +583,7 @@
     });
 
     app.controller("SharePostController", function SharePostController(user, posts, post, $mdDialog, PostService,
-     MessageService, $state) {
+        MessageService, $state, addpost) {
```
Comment (andreldsa) [inline app/post/postDetailsDirective.js]: coloca o addpost junto com as outras variáveis locais, no início.

File: app/post/postDetailsDirective.js [hunk 3]
```diff
@@ -620,13 +622,19 @@
             PostService.createPost(shareCtrl.newPost).then(function success(response) {
                 MessageService.showToast('Compartilhado com sucesso!');
                 $mdDialog.hide();
-                posts.push(response.data);
+                shareCtrl.addPostTimeline(response.data);
```
Comment (andreldsa) [inline app/post/postDetailsDirective.js]: teste

File: app/post/postDetailsDirective.js [hunk 4]
```diff
@@ -159,7 +159,8 @@
                 locals: {
                     user : postDetailsCtrl.user,
                     posts: postDetailsCtrl.posts,
-                    post: post
+                    post: post,
+                    addpost: postDetailsCtrl.addpost
```
Comment (andreldsa) [inline app/post/postDetailsDirective.js]: camelcase

File: app/home/home.html [hunk 5]
```diff
@@ -43,7 +43,7 @@
       </md-card>
     </div>
 
-    <post-timeline ng-if="!homeCtrl.isLoadingPosts" flex-gt-md="60" flex-md="75" layout="column" md-colors="{background: 'grey-200'}" posts="homeCtrl.posts" institution="false" user="homeCtrl.user"></post-timeline>
+    <post-timeline ng-if="!homeCtrl.isLoadingPosts" flex-gt-md="60" flex-md="75" layout="column" md-colors="{background: 'grey-200'}" posts="homeCtrl.posts" institution="false" user="homeCtrl.user" addpost="true" ></post-timeline>
```
Comment (andreldsa) [inline app/home/home.html]: quando for usar camelcase a string fica com ífen, addPost = add-post

File: app/test/specs/sharePostControllerSpec.js [hunk 6]
```diff
@@ -182,4 +182,25 @@
...
+        });
+        
+        it("shouldn't add post in posts", function() {
+            shareCtrl.addPost = false;
+            expect(shareCtrl.posts).not.toContain(newPost);
+
+            shareCtrl.addPostTimeline(newPost);
+
+            expect(shareCtrl.posts).not.toContain(newPost);
+
```
Comment (andreldsa) [inline app/test/specs/sharePostControllerSpec.js]: rlb

File: app/test/specs/sharePostControllerSpec.js [hunk 7]
```diff
@@ -182,4 +182,25 @@
...
+            expect(shareCtrl.posts).toContain(newPost);
+        });
+        
+        it("shouldn't add post in posts", function() {
+            shareCtrl.addPost = false;
+            expect(shareCtrl.posts).not.toContain(newPost);
+
+            shareCtrl.addPostTimeline(newPost);
+
```
Comment (andreldsa) [inline app/test/specs/sharePostControllerSpec.js]: rlb

**Pull request merged**